src/material/select/src/select.type.ts

```diff
@@ -30,10 +40,30 @@ import { FieldType } from '@ngx-formly/material/form-field';
 })
 export class FormlyFieldSelect extends FieldType {
   @ViewChild(MatSelect) formFieldControl!: MatSelect;
+
   defaultOptions = {
     templateOptions: { options: [] },
   };
 
+  get value() { return this.formControl.value || []; }
+  get state() {
+    if (this.value.length > 0) {
+      return this.value.length !== this.formFieldControl.options.length
+        ? 'indeterminate'
+        : 'checked';
+    }
+
+    return '';
+  }
+
+  toggleSelectAll() {
+    this.formControl.setValue(
+      this.value.length !== this.formFieldControl.options.length
+        ? this.formFieldControl.options.map(x => x.value)
```

Comment:
This doesn't work with option groups.
I think that would work:
```js
this.formFieldControl.options.map(
  x => (x.group ? x.group.map(grp => grp.value) : x.value)
).flat()
```

Reply:
@fontsie yep in fact it doesn't . Can u open a new issue. We should work on a PR to fix it